wallet/src/main/scala/org/bitcoins/wallet/WalletCallbacks.scala

```diff
@@ -0,0 +1,128 @@
+package org.bitcoins.wallet
+
+import org.bitcoins.core.api.{Callback, CallbackHandler}
+import org.bitcoins.core.protocol.transaction.Transaction
+import org.bitcoins.core.util.FutureUtil
+import org.bitcoins.wallet.models.{AddressDb, SpendingInfoDb}
+import org.slf4j.Logger
+
+import scala.concurrent.{ExecutionContext, Future}
+
+/**
+  * Callbacks for responding to events in the wallet.
+  * The appropriate callback is executed whenever the wallet finishes,
+  * the corresponding function.
+  */
+trait WalletCallbacks {
+
+  def onTransactionProcessed: CallbackHandler[
```

Comment:
Is there a reason these are `Transaction` and the other things are `AddressDb` and `SpendingInfoDb` (two callbacks is `Db` the others are not)

Reply:
Not particularly, we could give the `TransactionDb` but it doesn't have any information you can't calculate from just the `Transaction`

Reply:
I think we should be consistent with whatever we do, either `Db` or no `Db`.